src/current/v25.2/deploy-cockroachdb-with-kubernetes-operator.md

```diff
@@ -0,0 +1,705 @@
+        cockroachdb:
+          tls:
+            enabled: true
+            externalCertificates:
+              enabled: true
+              certificates:
+                nodeSecretName: {node_secret_name}
+                nodeClientSecretName: {client_secret_name}
+        ~~~
+        
+        Replacing the following:
+        - `{node_secret_name}`: The name of the Kubernetes secret that contains the generated client certificate and key.
+        - `{client_secret_name}`: The name of the Kubernetes secret that contains the generated node certificate and key.
+        
+        For a detailed tutorial of a TLS configuration with manual certificates, refer to [Example: Authenticate with cockroach cert](#example-authenticate-with-cockroach-cert).
+
+1. In `cockroachdb.crdbCluster.localityMappings`, provide [locality mappings](#localities) that define locality levels and map them to node labels where the locality information of each Kubernetes node is stored. When CockroachDB is initialized on a node, it processes these values as though they are provided through the [`cockroach start --locality`](cockroach-start#locality) flag. 
+
+    The default configuration uses the `region` and `zone` locality labels, mapped implicitly to the [`topology.kubernetes.io/region`](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesioregion) and [`topology.kubernetes.io/zone`](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone) node labels.
+    - In cloud provider deployments, the `topology.kubernetes.io/region` and `topology.kubernetes.io/zone` values on a node are populated by the cloud provider.
+    - In bare metal deployments, the `topology.kubernetes.io/region` and `topology.kubernetes.io/zone` node label values are not set implicitly by a cloud provider when initializing the node, so you must set them manually or configure custom locality labels.
+
+    To add more granular levels of locality to your nodes or use different locality labels, add custom locality levels as values in the `cockroachdb.crdbCluster.localityMappings` list. Any custom `localityMappings` configuration overrides the default `region` and `zone` configuration, so if you append an additional locality level but wish to keep the `region` and `zone` labels you must declare them manually.
+
+    The following example uses the existing `region` and `zone` labels and adds an additional `datacenter` locality mapping that is more granular than `zone`. This example declares that the `dc` locality information is stored in the `example.datacenter.locality` node label:
+
+    ~~~ yaml
+    cockroachdb:
+      crdbCluster:
+        localityMappings:
+          - nodeLabel: "topology.kubernetes.io/region"
+            localityLabel: "region"
+          - nodeLabel: "topology.kubernetes.io/zone"
+            localityLabel: "zone"
+          - nodeLabel: "example.datacenter.locality"
+            localityLabel: "dc"
+    ~~~
```

Comment:
Needs eng review (check that the localityMappings example and updated description is correct)